cmd/livesim2/app/livesegment.go

```diff
@@ -647,40 +647,36 @@ func findRefSegMeta(a *asset, cfg *ResponseConfig, segmentPart string, nowMS int
 	return refMeta, nil
 }
 
-// writeChunkedSegment splits a segment into chunks and send them as they become available timewise.
-//
-// nowMS servers as reference for the current time and can be set to any value. Media time will
-// be incremented with respect to nowMS.
-func writeChunkedSegment(ctx context.Context, log *slog.Logger, w http.ResponseWriter, cfg *ResponseConfig, drmCfg *drm.DrmConfig,
-	vodFS fs.FS, a *asset, segmentPart string, nowMS int, isLast bool) error {
-
-	log.Debug("writeChunkedSegment", "segmentPart", segmentPart)
+// prepareChunks generates a live segment, chunks it, and encrypts it if needed.
+func prepareChunks(log *slog.Logger, vodFS fs.FS, a *asset, cfg *ResponseConfig, drmCfg *drm.DrmConfig,
+	segmentPart string, nowMS int, isLast bool, chunkIndex *int) (segOut, []chunk, error) {
 
 	so, err := genLiveSegment(log, vodFS, a, cfg, segmentPart, nowMS, isLast)
 	if err != nil {
-		return fmt.Errorf("convertToLive: %w", err)
+		return so, nil, fmt.Errorf("convertToLive: %w", err)
+	}
+	if isImage(segmentPart) {
+		return so, nil, nil
 	}
 	if so.seg == nil {
-		return fmt.Errorf("no segment data for chunked segment")
+		return so, nil, fmt.Errorf("no segment data for chunked segment")
 	}
 
-	w.Header().Set("Content-Type", so.meta.rep.SegmentType())
-	if isImage(segmentPart) {
-		w.Header().Set("Content-Length", strconv.Itoa(len(so.data)))
-		_, err = w.Write(so.data)
-		return fmt.Errorf("could not write image segment: %w", err)
-	}
 	rep := so.meta.rep
 	seg := so.seg
 
-	// Some part of the segment should be available, and is delivered directly.
-	// The rest are returned HTTP chunks as time passes.
-	// In general, we should extract all the samples and build a new one with the right fragment duration.
-	// That fragment/chunk duration is segment_duration-availabilityTimeOffset.
-	chunkDur := (a.SegmentDurMS - int(cfg.AvailabilityTimeOffsetS*1000)) * int(rep.MediaTimescale) / 1000
-	chunks, err := chunkSegment(rep.initSeg, seg, so.meta, chunkDur)
+	// Calculate chunk duration in media timescale units
+	var chunkDur int
+	if cfg.ChunkDurS != nil && *cfg.ChunkDurS > 0 {
+		// Use explicit chunk duration from URL parameter
+		chunkDur = int(*cfg.ChunkDurS * float64(rep.MediaTimescale))
+	} else {
+		// Fallback: fragment/chunk duration is segment_duration-availabilityTimeOffset
```

Comment:
OK. Nice short-cut. It should be documented somewhere if there is such an implicit setting of chunk durration.